prov/sockets/src/sock_ep_msg.c

```diff
@@ -693,8 +694,10 @@ static void *sock_msg_ep_listener_thread(void *data)
 	poll_fds[0].events = poll_fds[1].events = POLLIN;
 
 	while (*((volatile int*) &ep->attr->cm.do_listen)) {
+		fastlock_acquire(&ep->attr->cm.lock);
 		timeout = dlist_empty(&ep->attr->cm.msg_list) ? -1 :
 					SOCK_CM_COMM_TIMEOUT;
+		fastlock_release(&ep->attr->cm.lock);
```

Comment:
I removed this commit from the PR.  But I think the "re-write the 'you should keep listening' mechanism" solution sounds like the right idea.  I do not have the availability to do this, however.  @shantonu @jithinjosepkl is this something you guys could do?